hazelcast/src/main/java/com/hazelcast/map/impl/operation/PartitionCheckIfLoadedOperation.java

```diff
@@ -72,7 +72,8 @@ public Object getResponse() {
 
     @Override
     public boolean returnsResponse() {
-        return !waitForKeyLoad;
+        // has to return true in all cases in order to enable the default retry invocation mechanism
+        return true;
```

Comment:
we can wholly remove this `returnsResponse` method from this class and can rely on its super. Super returns `true` always.